omod/src/main/java/org/openmrs/module/spa/web/SinglePageApplicationServlet.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.spa.web;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.InputStream;
+import java.util.Scanner;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class SinglePageApplicationServlet extends HttpServlet {
+	
+	private String htmlFile = "";
+	
+	private Log log = LogFactory.getLog(this.getClass());
+	
+	public SinglePageApplicationServlet() {
+		super();
+		try {
+			InputStream inputStream = SinglePageApplicationServlet.class.getResourceAsStream("/web/module/index.htm");
+			Scanner scanner = new Scanner(inputStream);
+			scanner.useDelimiter("\\Z");
+			this.htmlFile = scanner.next();
```

Comment:
So it seems like if this fails, you're returning an empty document with a 200?

Maybe we should initialize htmlFile to null (just `private String htmlfile;`), do a null check in `doGet`, and if null return a 500?

Reply:
I like it -- will change.